controllers/component_controller.go

```diff
@@ -291,8 +298,137 @@ func (r *ComponentReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		}
 	}
 
+	if component.Spec.Build.ContainerImage == "" {
```

Comment:
This entire logic from here until EOF should be enclosed in 

```
if component.Spec.Source.GitSource != nil && component.Spec.Source.GitSource.URL != "" {
		// logic for Trigger Template, Event Listener and Webhook Route
}
```

Reply:
Could we actually put the entire build logic in a separate function and call it from the reconcile function? My thinking is that the main reconcile loop for this function is already pretty long, and I'm wondering if it would help readability if we put the build functionality into its own function. 

Might also help with unit testing some of the edge cases in here, since testenv can be pretty inflexible in that regard

Reply:
I'm wondering too, in addition to extracting this out to a function as I mentioned, I wonder if we could move the the build logic up into the end of the if / else block up above? Basically just above [here](https://github.com/redhat-appstudio/application-service/blob/26c102883a080a099d6df30840844f6999d8350d/controllers/component_controller.go#L233) for Component create, and just above [here](https://github.com/redhat-appstudio/application-service/blob/26c102883a080a099d6df30840844f6999d8350d/controllers/component_controller.go#L295) for component update.

It would also enable us to set the proper status condition (e.g. Create/Update) upon success / failures (by calling r.SetCreateConditionAndUpdateCR` and `r.SetUpdateConditionAndUpdateCR`)

During Component cr